### PR TITLE
Expose utilities

### LINKED ISSUE #4320: Consider exporting 'getOwnEmitOutputFilePath'
Courtesy of @ivogabe in [Gitter on August 14, 2015 5:26 AM](https://gitter.im/Microsoft/TypeScript?at=55cdde93057d8c9d3a6d55d2):

> Question about the API, is there an exported method that returns the output filename of a source file? It should follow the options `outDir` & `rootDir` of course.

### PR DESCRIPTION
**Pull request by ivogabe**
I'm working on a project that analyzes TypeScript sources and creates files in the same directories as the generated .js files. I prefer to use functions from the TypeScript api instead of writing them myself, so can these functions be exported?

Fixes #4320

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -2155,6 +1676,475 @@ namespace ts {
         return options.target === ScriptTarget.ES6 ? "lib.es6.d.ts" : "lib.d.ts";
     }
 
+    // Gets the nearest enclosing block scope container that has the provided node
+    // as a descendant, that is not the provided node.
+    export function getEnclosingBlockScopeContainer(node: Node): Node {
```
Comment (DanielRosenwasser) [inline src/compiler/utilities.ts]: These are still in `utilities.ts`, so was there any reason they were moved around other than locality?

Comment (ivogabe) [inline src/compiler/utilities.ts]: The file has two namespace blocks, the first is marked @internal, the second not, so the first contains all internal methods and the second the exported.

Comment (DanielRosenwasser) [inline src/compiler/utilities.ts]: Gotcha, wasn't aware of that change.

**Comment by DanielRosenwasser** [comment]
I'm hesitant to expose many of these functions because it implies a commitment to support in our API. Can @CyrusNajmabadi and @mhegazy weigh in on this?

**Comment by mhegazy** [comment]
I agree. Hiding these APIs was intentional to limit the surface area. We will need to do another API review to make sure everyone is comfortable expanding the API. i would file an issue with the needed APIs first though.

**Pull request closed without merging**